### PR TITLE
Introduce Graph format to Ketcher

### PR DESCRIPTION
**Pull request by Aniuta907**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ unstable:
     - master
     - dev
     - /^task-.*$/
+    - FORMAT
```
Comment (Vesdet) [inline .gitlab-ci.yml:32]: Remove this file at all since we moved to GitHub

File: gulp/check.js [hunk 2]
```diff
@@ -32,3 +32,12 @@ module.exports.checkDepsExact = function (options, cb) {
 		cb();
 	}
 };
+
+module.exports.lint = function (options) {
```
Comment (Vesdet) [inline gulp/check.js:36]: https://github.com/epam/ketcher/pull/49/files#diff-2c34f4245be5d3938b03ed57dc1e45d3L53 - Artur removed gulp lint task in scope of his merged PR, verify with him that changes

File: package.json [hunk 3]
```diff
@@ -92,9 +95,12 @@
     "test": "npm run test-io && npm run test-render",
     "test-io": "gulp test-io && node test io/io",
     "test-render": "gulp test-render && node test render",
+    "test-jest": "jest --coverage=true",
```
Comment (Vesdet) [inline package.json:98]: Rename to test-unit and add this tests to "test" npm-script

File: package.json [hunk 4]
```diff
@@ -92,9 +95,12 @@
...
     "generate-svg": "gulp test-render && node test/utils/generate-svg.js",
     "style": "gulp style",
     "gulp": "gulp",
+    "postinstall": "guppy pre-commit || exit 0",
+    "test-watcher": "jest --watchAll",
```
Comment (Vesdet) [inline package.json:103]: Let's name it "test-unit:dev"

File: package.json [hunk 5]
```diff
@@ -104,6 +110,7 @@
     "font-face-observer": "1.0.0",
     "jsonschema": "1.1.1",
     "lodash": "4.17.19",
+    "molfile": "0.0.5",
```
Comment (Vesdet) [inline package.json:113]: ?? Do you use it?

File: src/script/chem/molfile/molfile.js [hunk 6]
```diff
@@ -208,7 +208,14 @@ Molfile.prototype.writeCTab2000Header = function () {
 
 	this.writePaddedNumber(0, 3);
 	this.writeWhiteSpace(3);
-	this.writePaddedNumber(this.molecule.isChiral ? 1 : 0, 3);
+	const isAbsFlag = Array.from(this.molecule.frags.values())
+		.some(fr => {
```
Comment (Vesdet) [inline src/script/chem/molfile/molfile.js:212]:
```suggestion
		.some(fr => fr ? fr.enhancedStereoFlag === 'abs' : false)
```

File: src/script/chem/molfile/v2000.js [hunk 7]
```diff
@@ -234,7 +234,7 @@ function parseCTabV2000(ctabLines, countsSplit) { // eslint-disable-line max-sta
 	const atomCount = utils.parseDecimalInt(countsSplit[0]);
 	const bondCount = utils.parseDecimalInt(countsSplit[1]);
 	const atomListCount = utils.parseDecimalInt(countsSplit[2]);
-	ctab.isChiral = !ctab.isChiral ? utils.parseDecimalInt(countsSplit[4]) !== 0 : true;
+	const isAbs = utils.parseDecimalInt(countsSplit[4]) === 1; // todo: check it
```
Comment (Vesdet) [inline src/script/chem/molfile/v2000.js:237]: What this TODO means? Please verify that it works properly and remove TODO

File: src/script/chem/molfile/v3000.js [hunk 8]
```diff
@@ -185,7 +185,7 @@ function parseCTabV3000(ctabLines, norgroups) { // eslint-disable-line max-state
 	if (ctabLines[shift].slice(0, 13) != 'M  V30 COUNTS')
 		throw Error('CTAB V3000 invalid');
 	var vals = ctabLines[shift].slice(14).split(' ');
-	ctab.isChiral = (utils.parseDecimalInt(vals[4]) == 1);
+	const isAbs = utils.parseDecimalInt(vals[4]) === 1; // todo: check it
```
Comment (Vesdet) [inline src/script/chem/molfile/v3000.js:188]: The same with previous

File: src/script/chem/struct/fragment.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
+/****************************************************************************
+ * Copyright 2018 EPAM Systems
```
Comment (Vesdet) [inline src/script/chem/struct/fragment.js:2]: Set Copyright to 2020 for new files

File: src/script/editor/operations/op.js [hunk 10]
```diff
@@ -114,6 +87,9 @@ function AtomDelete(aid) {
 		restruct.atoms.delete(this.data.aid);
 		restruct.markItemRemoved();
 
+		// TODO: - here
```
Comment (Vesdet) [inline src/script/editor/operations/op.js:90]: Investigate this moment please. Check when in called and what can be missed on that step

File: src/script/editor/tool/index.js [hunk 11]
```diff
@@ -34,6 +33,9 @@ import ReactionUnmapTool from './reactionunmap';
 import PasteTool from './paste';
 import RotateTool from './rotate';
 
+// TODO: correct
+import EnhancedStereoTool from './enhanced-stereo';
```
Comment (Vesdet) [inline src/script/editor/tool/index.js:37]: Looks like tool working, remove TODO and empty lines

File: src/script/index.js [hunk 12]
```diff
@@ -102,11 +103,26 @@ const buildInfo = {
...
-	showMolfile
+	showMolfile,
+
+	// TODO: remove it
+	toGraph: () => {
+		const j = graph.toGraph(ketcher.editor.render.ctab.molecule);
+		validateGraphF(j);
+		return j;
+	},
+	fromGraph: () => graph.fromGraph(graph.toGraph(ketcher.editor.render.ctab.molecule)),
```
Comment (Vesdet) [inline src/script/index.js:122]: Remove `loadStruct` API - was used as debug, replace `toGraph` with `getGraph` and write correct method for `setGraph`

File: src/script/ui/component/view/icon.jsx [hunk 13]
```diff
@@ -17,6 +17,9 @@
 import { h } from 'preact';
 
 function Icon({ name, ...props }) {
+	// TODO: correct
+	if (name === 'enhanced-stereo')
+		name = 'cip';
```
Comment (Vesdet) [inline src/script/ui/component/view/icon.jsx:22]: Please add svg file for this Icon and remove this statement

File: src/script/ui/app/toolbar.jsx [hunk 14]
```diff
@@ -169,7 +169,7 @@ const template = [
 		component: TemplatesList
 	},
 	'template-lib',
-	'chiral-flag'
+	'enhanced-stereo'
```
Comment (Vesdet) [inline src/script/ui/app/toolbar.jsx:172]: Please read in details #56 , may be we should move button to top toolbar or somewhere else? Not sure that enhanced stereo tool should be on the same place where was chiral-flag tool

File: src/script/chem/struct/fragment.js (hunk 9 above)
Comment (Aniuta907) [reply src/script/chem/struct/fragment.js:2]: Do you mean for new files in this branch or new files created by me?

File: .gitlab-ci.yml (hunk 1 above)
Comment (Aniuta907) [reply .gitlab-ci.yml]: Done

File: gulp/check.js (hunk 2 above)
Comment (Aniuta907) [reply gulp/check.js]: Done

File: package.json (hunk 3 above)
Comment (Aniuta907) [reply package.json]: Done

File: package.json (hunk 4 above)
Comment (Aniuta907) [reply package.json]: Done

File: src/script/chem/struct/fragment.js (hunk 9 above)
Comment (Vesdet) [reply src/script/chem/struct/fragment.js:2]: All files in branch

File: package.json (hunk 5 above)
Comment (Aniuta907) [reply package.json]: Deleted it

File: src/script/chem/molfile/v2000.js (hunk 7 above)
Comment (Aniuta907) [reply src/script/chem/molfile/v2000.js]: Done

File: src/script/chem/molfile/v3000.js (hunk 8 above)
Comment (Aniuta907) [reply src/script/chem/molfile/v3000.js]: Done

File: src/script/editor/tool/index.js (hunk 11 above)
Comment (Aniuta907) [reply src/script/editor/tool/index.js:35]: Done

File: src/script/chem/struct/fragment.js (hunk 9 above)
Comment (Aniuta907) [reply src/script/chem/struct/fragment.js]: Done

File: src/script/ui/component/view/icon.jsx (hunk 13 above)
Comment (Aniuta907) [reply src/script/ui/component/view/icon.jsx]: Done

File: src/script/ui/app/toolbar.jsx (hunk 14 above)
Comment (Aniuta907) [reply src/script/ui/app/toolbar.jsx:172]: We agreed to do it later

File: src/script/editor/operations/op.js (hunk 10 above)
Comment (Aniuta907) [reply src/script/editor/operations/op.js]: Deleted it

**Review by AndreiMazol — APPROVED** [review]

**Pull request merged**